code-challenges.test.js

```diff
@@ -15,35 +16,115 @@
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+
+describe('contentShuffler', () => {
+    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    it('takes in an array, removes the first item from the array and shuffles the remaining content.', () => {
+      expect(contentShuffler(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]));
+      expect(contentShuffler(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]));
+    })
+})
+
+// Successful Failure 
+//  FAIL  ./code-challenges.test.js
+//   contentShuffler
+//   ✕ takes in an array, removes the first item from the array and shuffles the remaining content. (1 ms)
+// ● contentShuffler › takes in an array, removes the first item from the array and shuffles the remaining content.
+//   ReferenceError: contentShuffler is not defined
 
 
 // b) Create the function that makes the test pass.
+// Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
+
+// Pseudo code 
+// create a function 
+// input is an array 
+// we need to first remove the first item in the array by using .shift()
+// then shuffle the remaining - using math.floor and math.random somewhere in our code to randomize the remaining items in the array 
+// use a for loop -- since initial approach of a HOF did not work. 
+
 
+const contentShuffler = (array) => {
+    array.shift()
+    for (i = 0; i < array.length; i++) {
+        let randomizer = Math.floor(Math.random() * array.length)
+        let newArray = array [i]
+        array[i] = array[randomizer]
+        array[randomizer] = newArray 
+    }
+    return array
+}
 
-// --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
+
+// // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
 
+describe('miniMax', () => {
+    const nums1 = [3, 56, 90, -8, 0, 23, 6]
+    const nums2 = [109, 5, 9, 67, 8, 24]
+    it('takes an array of numbers and returns an array of the minimum and maximum numbers in that order.', () => {
+      expect(miniMax(nums1)).toEqual(expect.arrayContaining([-8, 90]));
```

Comment:
This expect.arrayContaining was meant for the first challenge to allow the values to appear at random indexes. Not necessary for any other questions. Still good structure